hydromt/models/_v1/model_root.py

```diff
@@ -0,0 +1,187 @@
+"""the new model root class."""
+
+from glob import glob
+from logging import FileHandler, Logger, getLogger
+from os import PathLike, getcwd, getlogin, makedirs
+from os.path import abspath, dirname, isdir, join
+from pathlib import Path, PurePosixPath, PureWindowsPath
+from platform import system
+from typing import List, Optional, Tuple, cast
+
+from hydromt.log import add_filehandler
+from hydromt.typing import ModeLike, ModelMode
+
+logger = getLogger(__name__)
+
+
+class ModelRoot:
+    """A class to handle model roots in a cross platform manner."""
+
+    def __init__(
+        self,
+        path: PathLike,
+        mode: ModeLike = "w",
+        folders: Optional[List[str]] = None,
+        create_dirs: bool = False,
```

Comment:
I'm fine with removing the `create_dirs` flag and not creating folders by default, but we should not past lists as default arguments. This is refered to as "Dangerous defaults" in python. Let me know if you want me to explain in more detail but this should not be done.

Reply:
Since we decided to take this out of the model root and move it to the model components, I've removed all of these arrtibutes from the Root model. I think we can pick this conversation up again when I start work on the Model Components